### PR TITLE
🌐 Add German translation for `docs/de/docs/index.md`, `docs/de/docs/fastapi-people.md`, and `docs/de/docs/python-types.md`

### PR DESCRIPTION
**Pull request by 0x4Dark**
Hello,

I have started to translate the documentation into german (to be honest, I was a bit surprised that nothing existed here yet).
Currently I have translated the following chapters.
- index
- features
- fastapi-people
- python-types

I thought to translate them piece by piece just following the order of the mkdocs nav.
Since I have never worked with MkDocs, I hope the "setup" for the german section is correct.

_P.S.: Please don't be surprised about the short time intervals of the commits. No, I am not so fast in translating - It took me a while. I only had a bad setup near the end and had to reset everything. ._

### CODE REVIEW

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:45]:
```suggestion
<div class="user"><a href="{{ user.url }}" target="_blank"><div class="avatar-wrapper"><img src="{{ user.avatarUrl }}"/></div><div class="title">@{{ user.login }}</div></a> <div class="count">Antworten: {{ user.count }}</div></div>
```

This can be translated.

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:63]:
```suggestion
<div class="user"><a href="{{ user.url }}" target="_blank"><div class="avatar-wrapper"><img src="{{ user.avatarUrl }}"/></div><div class="title">@{{ user.login }}</div></a> <div class="count">Antworten: {{ user.count }}</div></div>
```

Same here

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:71]:
```suggestion
## Top-Mitwirkende

Hier sind die **Top-Mitwirkenden**. 👷
```

I think this is gramatically correct.

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:81]:
```suggestion
<div class="user"><a href="{{ user.url }}" target="_blank"><div class="avatar-wrapper"><img src="{{ user.avatarUrl }}"/></div><div class="title">@{{ user.login }}</div></a> <div class="count">Pull-Requests: {{ user.count }}</div></div>
```

Same here

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:73]:
```suggestion
Diese Benutzer haben [die meisten Pull-Requests erstellt](help-fastapi.md#create-a-pull-request){.internal-link target=_blank}, welche *merged* wurden.
```

Same here

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:89]:
```suggestion
## Top-Rezensenten
```

Same here, and I think "Bewerter" or something like that is more common, "Rezensenten" isn't a very common word.

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:105]:
```suggestion
<div class="user"><a href="{{ user.url }}" target="_blank"><div class="avatar-wrapper"><img src="{{ user.avatarUrl }}"/></div><div class="title">@{{ user.login }}</div></a> <div class="count">Rezenzionen: {{ user.count }}</div></div>
```

This can be translated.

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:117]:
```suggestion
### Goldsponsoren
```

Again, I think this is gramatically correct.

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:125]:
```suggestion
### Silbersponsoren
```

Grammar, translate "Silver"

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:136]:
```suggestion
### Bronzesponsoren
```

Grammar

File: docs/de/docs/fastapi-people.md
Comment (AaronDewes) [inline docs/de/docs/fastapi-people.md:163]:
```suggestion
Das Hauptziel dieser Seite ist es, die Bemühungen der Gemeinschaft hervorzuheben, anderen zu helfen.
```

Ziel is more common and seems a bit better here.

File: docs/de/docs/features.md
Comment (AaronDewes) [inline docs/de/docs/features.md:28]:
```suggestion
Es basiert alles auf standardkonformen **Python 3.6 Typ**-Deklarationen (dank Pydantic). Sie müssen keine neue Syntax lernen. Nur modernes Standard-Python.
```

Grammar

File: docs/de/docs/features.md
Comment (AaronDewes) [inline docs/de/docs/features.md:100]:
```suggestion
Aber standardmäßig funktioniert das alles **"einfach so"**.
```

**Review by AaronDewes — CHANGES_REQUESTED** [review]
Thanks for your translation! This isn't a full review, but a few suggestions, I'll do more later.

**Comment by tiangolo** [comment]
This is awesome @0x4Dark! :rocket: You have done a lot of work! :bow: :clap: 

I have to wait for a couple of reviews before merging it. And as it has translations for several files, it will probably be difficult for someone to translate them all at the same time.

---

Let me suggest, could you split it into several PRs?

One would be just the basic setup, without any translation, just the result of:

```console
$ python ./scripts/docs.py new-lang de
```

And then a new PR for each file translated. That way, even if there are some changes for some of them, maybe one of them is already perfect and can be approved and merged faster.

**Review by AaronDewes — CHANGES_REQUESTED** [review]
Just saw this PR again, had a few comments I forgot to submit still here

**Comment by 0x4Dark** [comment]
Hi all,
Unfortunately, had the last few months no time to continue working on it and then forgot. Too many projects (of course with FastAPI).

Thanks for the feedback @AaronDewes.

@tiangolo I can do that, no problem. I'll work the changes from AaronDewes and create the pull requests asap.

File: docs/en/mkdocs.yml
Comment (rogerbrinkmann) [inline docs/en/mkdocs.yml:193]: name: de - Deutsch

File: docs/de/docs/python-types.md
Comment (rogerbrinkmann) [inline docs/de/docs/python-types.md:222]: The section about `Union`is missing.
Suggestion:

#### Union

You can declare that a variable can be any of **several types**, for example, an `int` or a `str`.

In Python 3.6 and above (including Python 3.10) you can use the `Union` type from `typing` and put inside the square brackets the possible types to accept.

In Python 3.10 there's also an **alternative syntax** were you can put the possible types separated by a <abbr title='also called "bitwise or operator", but that meaning is not relevant here'>vertical bar (`|`)</abbr>.

=== "Python 3.6 and above"

    ```Python hl_lines="1  4"
    {!> ../../../docs_src/python_types/tutorial008b.py!}
    ```

=== "Python 3.10 and above"

    ```Python hl_lines="1"
    {!> ../../../docs_src/python_types/tutorial008b_py310.py!}
    ```

In beiden Fällen bedeutet dies, dass `item` ein `int` oder ein `str` sein kann.

File: docs/de/docs/python-types.md
Comment (rogerbrinkmann) [inline docs/de/docs/python-types.md:224]: This section changed quite a bit in the original:
Suggestion:

Man kann deklarieren, dass ein Wert einen Typ haben kann, wie `str`, aber auch, dass er `None` sein kann.

In Python 3.6 und höher (einschließlich Python 3.10) können Sie dies deklarieren, indem Sie `Optional` aus dem `typing` Modul importieren und verwenden.

```Python hl_lines="1 4"
{!../../../docs_src/python_types/tutorial009.py!}
```

Durch die Verwendung von `Optional[str]` anstelle von `str` kann der Editor Ihnen helfen, Fehler zu erkennen, bei denen Sie davon ausgehen, dass ein Wert immer `str` ist, obwohl er eigentlich auch `None` sein könnte.

`Optional[Irgendwas]` ist eigentlich eine Abkürzung für `Union[Irgendwas, None]`, sie sind gleichwertig.

Das bedeutet auch, dass Sie in Python 3.10 `Irgendwas | None` verwenden können:

=== "Python 3.6 und höher"

    ```Python hl_lines="1 4"
    {!> ../../../docs_src/python_types/tutorial009.py!}
    ```

=== "Python 3.6 und höher - alternativ"

    ```Python hl_lines="1 4"
    {!> ../../../docs_src/python_types/tutorial009b.py!}
    ```

=== "Python 3.10 und höher"

    ```Python hl_lines="1"
    {!> ../../../docs_src/python_types/tutorial009_py310.py!}
    ```

File: docs/de/docs/python-types.md
Comment (rogerbrinkmann) [inline docs/de/docs/python-types.md:233]: This section also changed very much in the original

File: docs/de/docs/python-types.md
Comment (rogerbrinkmann) [inline docs/de/docs/python-types.md:266]: This section differs from the original as well

**Review by rogerbrinkmann — CHANGES_REQUESTED** [review]
A few suggestions. Unfortunately the englisch original already adopted changes in Python 3.9 and 3.10, which are not contained in this translation.

**Comment by tiangolo** [comment]
I updated it with recent changes, there's no longer need to use and synchronize complex `mkdocs.yml` files. I also removed the files that were already translated in other PRs and resolved the conflicts.

Nevertheless, it would probably be easier and better to have these 3 files in 3 different PRs so that it's easier to review them, and once one of the files is fine and approved (even if the other two are not), I could merge that if it was in an isolated PR.

As I know you're busy, maybe others might want to copy some of the work done here in one of these files and make a new PR only for that one, and continue from there. 🤓

**Comment by tiangolo** [comment]
📝 Docs preview for commit dd3d8a5f2ac9bd1e1f1dfadf282a64643d96d689 at: https://649a06e4d1bbbc073560fbdb--fastapi.netlify.app

**Comment by tiangolo** [comment]
As this is now being handled in 3 independent PRs at https://github.com/tiangolo/fastapi/pull/9502, https://github.com/tiangolo/fastapi/pull/10287, https://github.com/tiangolo/fastapi/pull/10285, I'm gonna close this one. Thanks everyone! 🚀

**Pull request closed without merging**
